### PR TITLE
CHORE for [DPE-5095] move over code to be re-used by K8s into correct lib

### PR DESCRIPTION
**Pull request by MiaAltieri**
## Issue
K8s charm needs to implement get_invalid_integration_status

## Solution
Add it to a lib so it can be easily re-used

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -636,7 +636,7 @@ def _on_storage_detaching(self, event: StorageDetachingEvent) -> None:
     def _on_update_status(self, event: UpdateStatusEvent):
         # user-made mistakes might result in other incorrect statues. Prioritise informing users of
         # their mistake.
-        invalid_integration_status = self.get_invalid_integration_status()
+        invalid_integration_status = self.set_status.get_invalid_integration_status()
```
Comment (Gu1nness) [inline src/charm.py]:
```suggestion
        invalid_integration_status = self.status.get_invalid_integration_status()
```

**Review by Gu1nness — CHANGES_REQUESTED** [review]

File: lib/charms/mongodb/v0/set_status.py [hunk 2]
```diff
@@ -223,8 +223,22 @@ def prioritize_statuses(self, statuses: Tuple) -> StatusBase:
...
-def build_unit_status(mongodb_config: MongoConfiguration, unit_host: str) -> StatusBase:
+        if not self.charm.backups.is_valid_s3_integration():
+            return BlockedStatus(
+                "Relation to s3-integrator is not supported, config role must be config-server"
+            )
+
+        return self.charm.get_cluster_mismatched_revision_status()
+
+
+def build_unit_status(mongodb_config: MongoDBConfiguration, unit_host: str) -> StatusBase:
```
Comment (Gu1nness) [inline lib/charms/mongodb/v0/set_status.py]:
```suggestion
def build_unit_status(mongodb_config: MongoConfiguration, unit_host: str) -> StatusBase:
```

File: lib/charms/mongodb/v0/set_status.py [hunk 3]
```diff
@@ -4,9 +4,9 @@
 # See LICENSE file for licensing details.
 import json
 import logging
-from typing import Tuple
+from typing import Optional, Tuple
 
-from charms.mongodb.v1.mongodb import MongoConfiguration, MongoDBConnection
+from charms.mongodb.v1.mongodb import MongoDBConfiguration, MongoDBConnection
```
Comment (Gu1nness) [inline lib/charms/mongodb/v0/set_status.py]: In a former PR, you unified the configuration which is now named `MongoConfiguration`.
```suggestion
from charms.mongodb.v1.mongodb import MongoConfiguration, MongoDBConnection
```

**Review by Gu1nness — CHANGES_REQUESTED** [review]

**Review by Gu1nness — APPROVED** [review]

**Review by reneradoi — APPROVED** [review]

**Review by juditnovak — APPROVED** [review]
LGTM

**Pull request merged**